File: E-EstateV2 API/E-EstateV2 API/Models/Field.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace E_EstateV2_API.Models
{
    public class Field
    {
        [Key]
        public int Id { get; set; }
        public string fieldName { get; set; }
        public float area { get; set; }
        public bool isMature { get; set; }
        public bool isActive { get; set; }
        public DateTime? dateOpenTapping { get; set; }
        public int yearPlanted { get; set; }
        public int totalTask { get; set; }
        public int initialTreeStand { get; set; }
        public string createdBy { get; set; }
        public DateTime createdDate { get; set; }
        public string updatedBy { get; set; }
        public DateTime updatedDate { get; set; }

        [ForeignKey("FieldStatusId")]
        public int? fieldStatusId { get; set; }
        public FieldStatus FieldStatus { get; set; }
        public int estateId { get; set; }

    }
}

File: E-EstateV2/src/app/_interface/field.ts
import { Clone } from "./clone";
import { FieldStatus } from "./fieldStatus";

export interface Field {
    percentage: number;
    id: number
    fieldName: string
    area: number
    isActive: boolean
    isMature: boolean | null
    dateOpenTapping: string | null
    dateOpenTappingFormatted: string
    totalTask: number
    estateId: number
    cloneId: number
    clone: string
    fieldStatusId: number
    fieldStatuses: FieldStatus[]
    fieldStatus: string
    yearPlanted: string
    otherCrop: string
    cloneName: string
    clones: Clone[]
    initialTreeStand: number
    tappingSystem: string
    createdBy: string
    createdDate: Date
    updatedBy: string
    updatedDate: Date
    conversionCropName: string
    sinceYear: number | null
    fieldId: number
    conversionId: number
    fieldDiseaseId:number
    areaInfected:number
    estate:any
    otherCropId:number
    companyId:number
    rubberArea:number | null
    remark:string
    currentTreeStand:number
    
}